python_scripts/datasets_bike_rides.py

```diff
@@ -178,10 +178,16 @@
 indices = rng.choice(np.arange(cycling_ride.shape[0]), size=500, replace=False)
 
 # %%
+import warnings
```

Comment:
Here, we can avoid the warning by taking directly an explicit copy of the subset to indicate pandas that we are fine working with such copy.

so only change the following line:

```python
subset = cycling_ride.iloc[indices].copy()
```